docker/puppetdb/ssl-setup.sh

```diff
@@ -0,0 +1,454 @@
+#!/bin/sh
+
+ssl_command="puppetdb ssl-setup"
+
+#############
+# FUNCTIONS #
+#############
+
+# Display usage information and exit
+#
+# This function simply displays the usage information to the screen and exits
+# with exit code 0.
+#
+# Example:
+#
+#   usage
+#
+function usage() {
+  echo "Usage: ${ssl_command} [-if]"
+  echo "Configuration helper for enabling SSL for PuppetDB."
+  echo
+  echo "This tool will attempt to copy the necessary Puppet SSL PEM files into "\
+       "place for use by the PuppetDB HTTPS service. It also is able to update "\
+       "the necessary PuppetDB configuration files if necessary to point to "\
+       "the location of these files and also configures the host and port for "\
+       "SSL to listen on."
+  echo
+  echo "Options:"
+  echo " -i  Interactive mode"
+  echo " -f  Force configuration file update. By default if the configuration "\
+       "already exists in your jetty.ini or if your configuration is otherwise "\
+       "in a state we believe we shouldn't touch by default, you must use this "\
+       "option to override it"
+  echo " -h  Help"
+  exit 0
+}
+
+# Backs up a file, if it hasn't already been backed up
+#
+# $1 - file to backup
+#
+# Example:
+#
+#   backupfile "/etc/myconfig"
+#
+function backupfile() {
+  # Create the global array if it doesn't already exist
+  if [ -z $backupfile_list ]; then
+    # backupfile_list=()
+    backupfile_list=""
```

Comment:
Arrays aren't supported in `sh` but we can simulate them with a space-separated string.

I had to do this in 5 or 6 places in this script.

I left the original lines commented out so you can see the diff.